code/planning/src/planning/preplanning_trajectory.py

```diff
@@ -0,0 +1,1450 @@
+                                                     point3_x, point3_y,
+                                                     direction, width)
+                    dist = help_functions.euclid_dist(ref_point, point)
+                    # Choose the width value for the correct lane
+                    if dist < min_dist:
+                        min_dist = dist
+                        min_width = width
+                # if we need to change the lane for the next action
+                # we want a smooth change in the trajectory points
+                points = reference_line
+                last_width = widths[size-1]
+                diff = abs(min_width - last_width)
+                step_size = STEP_SIZE
+                i = 0
+                # if diff != 0 we need to change the lane and calculate
+                # a smooth change
+                if diff != 0:
+                    steps = int(diff / step_size)
+                    first_widths = [last_width - step_size * i
+                                    for i in range(steps)]
+                    for i in range(len(first_widths)):
+                        p1_x = points[0][i]
+                        p1_y = points[1][i]
+                        p2_x = points[0][i+1]
+                        p2_y = points[1][i+1]
+                        if i != len((points[0])) - 2:
+                            p3_x = points[0][i + 2]
+                            p3_y = points[1][i + 2]
+                        point, v = self.update_one_point(p1_x, p1_y,
+                                                         p2_x, p2_y,
+                                                         p3_x, p3_y,
+                                                         direction,
+                                                         first_widths[i])
+                        points[0][i] = point[0]
+                        points[1][i] = point[1]
+                    p = [points[0][i+1:], points[1][i+1:], points[2][i+1:],
+                         points[3][i+1:]]
+                    points = self.update_points(p, direction, min_width)
+                del points[0][index+1:]
+                del points[1][index+1:]
+                del points[2][index+1:]
+                del points[3][index+1:]"""
+
+
+""" find the hdg values for the endpoints of the possible roads
+        for road_id in possible_road_ids:
+            start, end = self.get_endpoints(road_id)
+            dist_start = help_functions.euclid_dist(start, last_point)
+            dist_end = help_functions.euclid_dist(end, last_point)
+            if dist_start < dist_end:
+                hdg = self.geometry_data[road_id][2][-1]
+                hdg = self.rad_to_degree(hdg)
+                yaw_values.append(hdg)
+            else:
+                hdg = self.geometry_data[road_id][2][0]
+                hdg = self.rad_to_degree(hdg)
+                yaw_values.append(hdg)
+        print(yaw_values)
+"""
```

Comment:
this should probably be deleted

Reply:
I commented it, because maybe i need it later if something is not working in future towns